#!/usr/bin/env python
PACKAGE = "camera_publish"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("exposureTime",    int_t,    0, "An Integer parameter", 1500,  1, 120000)
gen.add("skippedImage",    int_t,    0, "An Integer parameter", 5,  1, 30)
#gen.add("double_param", double_t, 0, "A double parameter",    .5, 0,   1)
gen.add("folder_name",    str_t,    0, "A string parameter",  "field test")
gen.add("use_input_folder_name",   bool_t,   0, "A Boolean parameter",  False)
gen.add("create_new_folder_and_save_image",   bool_t,   0, "A Boolean parameter",  False)
gen.add("use_old_folder_and_save_image",   bool_t,   0, "A Boolean parameter",  False)

#size_enum = gen.enum([ gen.const("Small",      int_t, 0, "A small constant"),
#                       gen.const("Medium",     int_t, 1, "A medium constant"),
#                       gen.const("Large",      int_t, 2, "A large constant"),
#                       gen.const("ExtraLarge", int_t, 3, "An extra large constant")],
#                     "An enum to set size")

#gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "camera_publish", "Tutorials"))
